File: arcade/python/arcade-intro/09_Dark Wilderness/041_digitDegree.py
def digitDegree(n):
    '''
    Let's define digit degree of some positive integer as the number of times 
    we need to replace this number with the sum of its digits until we get 
    to a one digit number.
    Given an integer, find its digit degree.
    '''
    count = 0
    while n // 10 > 0:
        n = sumDigits(n)
        count += 1
    return count

def sumDigits(n):
    sum = 0
    while n > 0:
        sum += n % 10
        n = n // 10
    return sum

File: arcade/go/arcade-intro/09_Dark Wilderness/041_digitDegree.go
func digitDegree(n int) int {
	degree := 0
	sum := n
	for sum >= 10 {
		s := strconv.Itoa(sum)
		sum = 0
		for _, r := range s {
			i := int(r - '0')
			sum += i
		}
		degree++
		fmt.Println(sum)
	}

	return degree
}
